definitions/npm/fs-extra_v7.x.x/flow_v0.104.x-/fs-extra_v7.x.x.js

```diff
@@ -1,773 +1,832 @@
+  declare function utimes(
     path: string | Buffer,
     atime: number,
     mtime: number,
     callback: (err: ErrnoError) => void
   ): void;
-  declare export function utimes(
+  declare function utimes(
     path: string | Buffer,
     atime: Date,
     mtime: Date,
     callback: (err: ErrnoError) => void
   ): void;
-  declare export function utimes(
+  declare function utimes(
     path: string | Buffer,
     atime: number,
     mtime: number
   ): Promise<void>;
-  declare export function utimes(
+  declare function utimes(
     path: string | Buffer,
     atime: Date,
     mtime: Date
   ): Promise<void>;
 
-  declare export function write(
+  declare function write(
     fd: number,
     buffer: Buffer,
     offset: number,
     length: number,
     position: number | null,
     callback: (err: ErrnoError, written: number, buffer: Buffer) => void
   ): void;
-  declare export function write(
+  declare function write(
     fd: number,
     buffer: Buffer,
     offset: number,
     length: number,
     callback: (err: ErrnoError, written: number, buffer: Buffer) => void
   ): void;
-  declare export function write(
+  declare function write(
     fd: number,
     data: any,
     callback: (err: ErrnoError, written: number, str: string) => void
   ): void;
-  declare export function write(
+  declare function write(
     fd: number,
     data: any,
     offset: number,
     callback: (err: ErrnoError, written: number, str: string) => void
   ): void;
-  declare export function write(
+  declare function write(
     fd: number,
     data: any,
     offset: number,
     encoding: string,
     callback: (err: ErrnoError, written: number, str: string) => void
   ): void;
-  declare export function write(
+  declare function write(
     fd: number,
     buffer: Buffer,
     offset: number,
     length: number,
     position?: number | null
   ): Promise<WriteResult>;
-  declare export function write(
+  declare function write(
     fd: number,
     data: any,
     offset: number,
     encoding?: string
   ): Promise<WriteResult>;
 
-  declare export function writeFile(
+  declare function writeFile(
     file: string | Buffer | number,
     data: any,
     callback: (err: ErrnoError) => void
   ): void;
-  declare export function writeFile(
+  declare function writeFile(
     file: string | Buffer | number,
     data: any,
     options?: WriteFileOptions | string
   ): Promise<void>;
-  declare export function writeFile(
+  declare function writeFile(
     file: string | Buffer | number,
     data: any,
     options: WriteFileOptions | string,
     callback: (err: ErrnoError) => void
   ): void;
 
-  declare export function mkdtemp(prefix: string): Promise<string>;
-  declare export function mkdtemp(
+  declare function mkdtemp(prefix: string): Promise<string>;
+  declare function mkdtemp(
     prefix: string,
     callback: (err: ErrnoError, folder: string) => void
   ): void;
+
+  declare module.exports: {
```

Comment:
By removing all the `export` and using `module.exports` you change this libdef from ES modules to commonjs. It should ba avoided when possible.

Reply:
I totally agree with you! But I have no idea how to combine type spreading and ES Module.

Reply:
Untested: `declare export default` ?

Reply:
```js
declare export default $ReadOnly<{|
  ...fsTypes;
  access: typeof access;
  appendFile: typeof appendFile;
  chmod: typeof chmod;
  // [...]
|}
```

Reply:
I remembered that the problem was with this notation approach I get errors for the next code:
```javascript
if( ... ) {
    const fs = require('fs-extra');
    const data = fs.readFileSync(argsFileName, { encoding: 'utf8' });
...
```
`// Error: Cannot call fs.readFileSync because property readFileSync is missing in  module 'fs-extra' [1].`

I would also add that `fs-extra` itself uses commonjs  export format.